### ISSUE TITLE
Pinterest oEmbed fails with .co.uk TLD

### ISSUE THREAD

**Issue by gemmagarner** [Automattic/jetpack#7968]
Inserting a Pinterest URL containing a .co.uk TLD does not embed a board, profile or individual pins.

#### Steps to reproduce the issue

1. Go to [pinterest.co.uk](https://www.pinterest.co.uk/)
2. Copy the URL of a board containing a Pinterest .co.uk TLD. For example:

`https://www.pinterest.co.uk/stayandroam/blue-and-white/
`

3. Go to [WordPress.com](https://wordpress.com/)
4. Go to Blog Posts > Add New
5. Click on the Visual editor tab
6. Paste in the URL from step 2 (.co.uk TLD)
7. Publish the post
8. View the post on the front end

#### What I expected

I expected to see the Pinterest board embedded in my post.

#### What happened instead

The Pinterest board was displayed as a URL rather than embedded into the post. However, I was able to successfully embed the Pinterest board by changing the Pinterest TLD from .co.uk to .com:

`https://www.pinterest.com/stayandroam/blue-and-white/
`

Once the URL had been updated and the post was saved, the Pinterest board was embedded.

#### Screenshot

<img width="826" alt="pinterest-embed" src="https://user-images.githubusercontent.com/11627765/31435405-5359231c-ae77-11e7-886a-e78e43fe194b.png">

#### Code

pinterest.com appears to be hard coded here:

https://github.com/Automattic/jetpack/blob/e0c3d2d4abb857e794985acfc762b9f867151065/modules/shortcodes/pinterest.php#L10

Reported in 879741-f

**Comment by lizkarkoski** [comment]
There's an additional report of this for the post https://diywardrobe.co.uk/2017/11/07/curated-closet-part-2-inspiration/ which was reported in the forum thread at http://en.forums.wordpress.com/topic/cant-embed-pinterest-board

**Comment by gemmagarner** [comment]
I've come across the same issue with .pt domains in 892882-f. I tested embedding with that extension and it did not work.

**Comment by darcie** [comment]
I saw the topic on this in the forums and came here, as that topic was closed. If you've only added .co.uk, there will still be outstanding Pinterest users that come back about the inability to embed. Pinterest shifted from using uk.pinterest.com to www.pinterest.co.uk and other similar tlds a few months ago. So .co.uk works now to embed on WordPress.com in addition to pinterest.com, but .ca or .de does not. 

As I understand, [Pinterest is going to keep adding TLDs](https://medium.com/@Pinterest_Engineering/how-switching-our-domain-structure-unlocked-international-growth-e00c8184d5dd), and so there are still more to add to embeds and will likely be more in the future.
